src/esmac_diags/preprocessing/prep_ACEENA_E3SM.py

```diff
@@ -1,1286 +1,1442 @@
-    NCNall.data = NCNall.data * 1e-6
-    NCNall.attrs['units']='#/cm3'
-    for vv in ['NCN3', 'NCN10', 'NCN100', 'Nd_mean', 'Nd_ARM', 'Nd_VISST']:
-        variables[variable_names.index(vv)].data = variables[variable_names.index(vv)].data * 1e-6
-        variables[variable_names.index(vv)].attrs['units']='#/cm3'
-    # LWC and IWC
-    for vv in ['TGCLDIWP','TGCLDLWP']:
-        variables[variable_names.index(vv)].data = variables[variable_names.index(vv)].data *1000
-        variables[variable_names.index(vv)].attrs['units']='g/m2'
-    # cloud fraction
-    for vv in ['CLDTOT','CLDLOW','CLDMED','CLDHGH']:
-        variables[variable_names.index(vv)].data = variables[variable_names.index(vv)].data *100
-        variables[variable_names.index(vv)].attrs['units']='%'
-    
-    #%% re-shape the data into pre-defined resolution
-    variables_new = list()
-    #1d variable. only numpy.interp can keep some single-point values (see Nd_mean)
-    for var in variables:
-        var_new = np.interp(np.int64(time_new), np.int64(e3smtime), var, left=np.nan, right=np.nan)
-        variables_new.append(var_new)
-    # treat variables with other dimensions (e.g., size distribution)    
-    f = interp1d(np.int64(e3smtime), NCNall, bounds_error=False)
-    NCNall_new = f(np.int64(time_new))
-    
-    # %% output extacted file
-    varall_1d = {
-            variable_names[vv]: ('time', np.float32(variables_new[vv])) for vv in range(len(variable_names))
-    }
-    varall_2d = {
-            'NCNall': (['size','time',], np.float32(NCNall_new))
-    }
-    varall_1d.update(varall_2d)
-    outfile = output_path + output_filehead + '_sfc.nc'
-    print('output file '+outfile)
-    ds = xr.Dataset( varall_1d,
-                      coords={'time': ('time', time_new), 'size':('size', np.arange(1,3001))})
-    #assign attributes
-    ds['time'].attrs["long_name"] = "Time"
-    ds['time'].attrs["standard_name"] = "time"
-    for vv in range(len(variable_names)):
-        ds[variable_names[vv]].attrs["long_name"] = variables[vv].long_name
-        ds[variable_names[vv]].attrs["units"] = variables[vv].units
-        ds[variable_names[vv]].attrs["description"] = "variables at surface, TOA or the lowest model level"
-    ds['NCNall'].attrs["long_name"] = NCNall.long_name
-    ds['NCNall'].attrs["units"] = NCNall.units
-    ds['NCNall'].attrs["description"] = 'calculated from modal information into 1nm increment'
-    
-    ds.attrs["title"] = 'preprocessed E3SM data at surface, TOA or the lowest model level'
-    ds.attrs["inputfile_sample"] = lst[0].split('/')[-1]
-    ds.attrs["date"] = ttt.ctime(ttt.time())
-    
-    ds.to_netcdf(outfile, mode='w')
+"""
+prepare E3SM output data 
+for surface (include remote sensing and satellite) and aircraft measurements in ACEENA campaign
+input data is regional hourly output data in E3SM
+all variables has a region appendix similar to "lat_330e_to_335e_37n_to_42n"
+"""
+
+import glob
+import os
+import re
+import fnmatch
```

Comment:
can you add fnmatch to the `environment.yml `and `requirements.txt` at the main directory of ESMAC Diags?

Reply:
Added it to the `environment.yml`.

Reply:
Removed it in commit 3. Using the original `environment.yml` now.